### ISSUE TITLE
moderately long path + urlsegment leads to 404 not found

### ISSUE THREAD

**Issue by er314** [ryancramerdesign/ProcessWire#1845]
in this request, 
`http://127.0.0.1/dir-level1-abcdefghijklmnopqrstuvwxyz-abcd/dir-level2-abcdefghijklmnopqrstuvwxyz-abcdefghijklmnopqrstuv/preview/`
`preview` is the url segment, and the rest is a valid path to a page.
`-> 404 not found.
`
after a debugging session, here is the reason : 
PW seeks for the real URL (out of url1/.../urlN/segm1/.../segmN/ ), and in order to do so, it first tries to find() a page with the full path, then if not found, it removes the right-most segment, then it **sanititizes** the remaining path, then it reiterates the find() on the remaining path, and so on.
The sanitization of the remaining path `$it` does the following, in `ProcessPageView.module` : 
```
$selector = "path=" . $this->wire('sanitizer')->selectorValue($it) . ", status<" . Page::statusMax;
$page = $this->pages->get($selector);
```
with
```
	public function selectorValue($value, $maxLength = 100) {
	[...]
	// selector value is limited to 100 chars
	if(strlen($value) > $maxLength) {
		if($this->multibyteSupport) $value = mb_substr($value, 0, $maxLength, 'UTF-8'); 
			else $value = substr($value, 0, $maxLength); 
	}
```
As a result, the remaining path for the next iteration of find() is truncated to 100 chars.
And of course it fails to find the page.
This is with PW 3.0.15.

I think it's not uncommon to have URLs (much) larger than 100 chars.
According to various sources, limitations on the browsers side and on the servers side vary between 2k and 8k.

**Comment by er314** [comment]
update : after looking at the code history, this is a regression which has been introduced in PW 3.0.12, with latest commit to `ProcessPageView.module`.
previously, instead of re-sanitizing the remaining path `$it`, the code was just doing : 
`$page = $this->pages->get("path=$it, status<" . Page::statusMax);`

**Comment by ryancramerdesign** [comment]
Thanks @er314, I'm always impressed how far you research these things and are able to track them down. That selectorValue sanitizer was added when we added support for UTF-8 page names, just in case someone forgot to define a whitelist for their page names. I've bumped up the length that it will support to 2048. While URLs can get longer, it usually has to do with a query string rather than a path.
